File: applications/event-account-gemfire-sink/src/test/java/showcase/event/stream/rabbitmq/account/AccountConsumerTest.java
package showcase.event.stream.rabbitmq.account;

import com.vmware.tanzu.data.services.rabbitmq.streaming.account.domain.Account;
import nyla.solutions.core.patterns.creational.generator.JavaBeanGeneratorCreator;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;
import showcase.event.stream.rabbitmq.account.functions.AccountConsumer;
import showcase.event.stream.rabbitmq.account.repostories.AccountRepository;

import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.verify;

@ExtendWith(MockitoExtension.class)
class AccountConsumerTest {

    private Account account = JavaBeanGeneratorCreator.of(Account.class).create();
    private AccountConsumer subject;
    @Mock
    private AccountRepository repository;

    @BeforeEach
    void setUp() {
        subject = new AccountConsumer(repository);
    }

    @Test
    void accept() {
        subject.accept(account);

        verify(repository).save(any());
    }
}

File: applications/account-geode-rabbit-streams-sink/src/test/kotlin/com/vmware/tanzu/data/services/rabbitmq/streaming/account/consumer/AccountConsumerTest.kt
package com.vmware.tanzu.data.services.rabbitmq.streaming.account.consumer

import com.vmware.tanzu.data.services.rabbitmq.streaming.account.domain.Account
import com.vmware.tanzu.data.services.rabbitmq.streaming.account.repository.AccountRepository
import nyla.solutions.core.patterns.creational.generator.JavaBeanGeneratorCreator
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.mockito.kotlin.any
import org.mockito.kotlin.mock
import org.mockito.kotlin.verify

internal class AccountConsumerTest {

    private lateinit var accountRepository: AccountRepository
    private lateinit  var account : Account

    @BeforeEach
    internal fun setUp() {
        accountRepository = mock<AccountRepository>()
        account = JavaBeanGeneratorCreator.of(Account::class.java).create()
    }

    @Test
    fun accept() {
        account = JavaBeanGeneratorCreator.of(Account::class.java).create()
        var subject = AccountConsumer(accountRepository)
        subject.accept(account)

        verify(accountRepository).save(any<Account>())
    }
}